packages/peregrine/lib/hooks/useScrollIntoView.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+
+/**
+ * Scrolls a ref into view on truthiness of a thing.
+ *
+ * @param {React.Ref} ref
+ * @param {Any} thing triggers scrolling if truthy
+ * @param {Object} options https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView
+ */
+export const useScrollIntoView = (
+    ref,
+    thing,
+    options = {
+        behavior: 'smooth',
+        block: 'center'
+    }
+) => {
+    useEffect(() => {
+        if (
+            ref.current &&
+            typeof ref.current.scrollIntoView !== 'undefined' &&
```

Comment:
Big improvement adding this check, thank you 🙂  Since this hook isn't responsible for ref creation and attachment, I'm curious if we need to be more strict. eg. `ref.current instanceof HTMLElement`. This would error if somehow a truthy ref made it in.

Reply:
![](https://media.giphy.com/media/4EiGNSTfy4WC4/giphy.gif)

Reply:
b518f7d